site/src/sphinx/server-decorator.rst

```diff
@@ -217,6 +217,37 @@ Therefore, you need to specify the decorators as extra parameters:
 A good real-world example of :api:`ServiceWithRoutes` is :api:`GrpcService`.
 See :ref:`server-grpc-decorator` for more information.
 
+Decorating a service by path mapping
+------------------------------------
+
+If you want to decorate a :api:`Service` by path mapping or router matching, you can specify a way to select
+the services which are being decorated with ``decorateUnder(pathPrefix, ...)`` or ``decorate(Route, ...)``.
+
+.. code-block:: java
+
+    import com.linecorp.armeria.common.HttpHeaderNames;
+
+    UserService userService = ...;
+    ServerBuilder sb = new ServerBuilder();
+    sb.annotatedService("/users/{id}", userService)
+      .decorateUnder("/users", (delegate, ctx, req) -> {
+          if (!authenticate(req)) {
+              return HttpResponse.of(HttpStatus.UNAUTHORIZED);
+          }
+          return delegate.serve(ctx, req);
+      })
+      .serviceUnder("/public", (ctx, req) -> {
+         // Serve public resources
+      })
+      // Use fluent route builder to match services being decorated
+      .routeDecorator()
+      .prefix("/public")
```

Comment:
How about separating this example into two examples? One that uses the traditional API and the other that uses the fluent route API? We could use the same path mappings (`/users`) to show the differences.

Reply:
Updated the documentation. I wonder it is suitable for this case. 😀